java/src/com/ctre/phoenix/motorcontrol/can/WPI_TalonSRX.java

```diff
@@ -43,9 +45,26 @@ public WPI_TalonSRX(int deviceNumber) {
 	@Override
 	public void set(double speed) {
 		_speed = speed;
-		set(ControlMode.PercentOutput, _speed);
+		if(mode == ControlMode.Velocity) {
+			set(wpiControlMode, _speed*maxVelocity);
+		} else {
+			set(wpiControlMode, _speed);
+		}
+		
 		_safetyHelper.feed();
 	}
+	
+	public void setWpiControlMode(ControlMode mode){
```

Comment:
WPI is an acronym and should be capitalized

Reply:
Also including WPI is confusing

Reply:
This isn’t changed when you use SRX libs natively; I thought this more clearly distinguished  that this is the only way to set mode for the WPI interface.

Reply:
Well it's only available in the WPI wrapper, so it's not necessary to specify that